### PR TITLE
pg-query-stream@3.0 release

### PR DESCRIPTION
**Pull request by brianc**
As mentioned in #2058 and #2051...I delayed releasing this until I could type again after 🔪 my finger on new years eve 🎆 😢 .  I noticed when typing up the changelog that I unintentionally entirely removed the ability to control the pg-query-stream's highWaterMark through the "proper" node readablestream config value so I added that back in with tests.  

Now that I can (mostly) type okay....let's get this out!  Immediately following this release today I'm going to put up a patch to fix a regression mentioned in #2056.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ For richer information consult the commit log on github with referenced pull req
 
 We do not include break-fix version release in this file.
 
+### pg-query-stream@3.0.0
+
+- [Rewrote stream internals](https://github.com/brianc/node-postgres/pull/2051) to better conform to node stream semantics.  This should make pg-query-stream much better at respecting [highWaterMark](https://nodejs.org/api/stream.html#stream_new_stream_readable_options) and getting rid of some edge case bugs when using pg-query-stream as an async iterator.  Due to the size and nature of this change (effectively a full re-write) it's safest to bump the semver major here, though almost all tests remain untouched and still passing, which brings us to a breaking change to the API....
+- Changed `stream.close` to `stream.destroy` which is the [official](https://nodejs.org/api/stream.html#stream_readable_destroy_error) way to terminate a readable stream.  This is a __breaking change__ if you rely on the `stream.close` method on pg-query-stream...though should be just a find/replace type operation to upgrade as the semantics remain very similar (not exactly the same, since internals are rewritte, but more in line with how streams are "supposed" to behave).
```
Comment (charmander) [inline CHANGELOG.md]: Minor typo fix:

```suggestion
- Changed `stream.close` to `stream.destroy` which is the [official](https://nodejs.org/api/stream.html#stream_readable_destroy_error) way to terminate a readable stream.  This is a __breaking change__ if you rely on the `stream.close` method on pg-query-stream...though should be just a find/replace type operation to upgrade as the semantics remain very similar (not exactly the same, since internals are rewritten, but more in line with how streams are "supposed" to behave).
```

**Comment by brianc** [comment]
Thanks for the reviews my friend! ❤️ Hope your new years was bright and all yer fingers stayed in 1 piece!
